soc/arm/nuvoton_npcx/common/reg/reg_def.h

```diff
@@ -505,17 +499,21 @@ struct adc_reg {
 	volatile uint8_t reserved3[2];
 	/* 0x026: Internal register 3 for ADC Speed */
 	volatile uint16_t MEAST;
-	volatile uint8_t reserved4[18];
-	/* 0x03A: Deassertion Threshold Control 1 Word */
-	volatile uint16_t THR_DCTL1;
-	/* 0x03C: Deassertion Threshold Control 2 Word */
-	volatile uint16_t THR_DCTL2;
-	/* 0x03E: Deassertion Threshold Control 3 Word */
-	volatile uint16_t THR_DCTL3;
-	/* 0x040 - 52: Data Buffer of Channel 0 - 9 */
-	volatile uint16_t CHNDAT[10];
 };
 
+static inline uint32_t npcx_thrctl_offset(uint32_t ctl_no)
+{
+	return DT_PROP(DT_NODELABEL(adc0), threshold_reg_offset) + (ctl_no - 1) * 2;
```

Comment:
why is this hard-coded to the nodelabel adc0?

Reply:
NPCX series only has one ADC hardware instance. NPCX7/9 used adc0 for the ADC node.
How about change to `DT_PROP(DT_INST(0, nuvoton_npcx_adc), threshold_reg_offset)`?